src/validator.cpp

```diff
@@ -258,6 +258,13 @@ struct Validator::ValidatorImpl
     void checkUnitForCycles(const ModelPtr &model, const UnitsPtr &parent,
                             std::vector<std::string> &history,
                             std::vector<std::vector<std::string>> &errorList);
+
+    /**
+    * @brief checks model mapping to ensure all variables mapped to other variables have the same units.
+    *
+    * @param model The model containing the variables and mappings to be tested.
+    */
+    void checkUnitHomogeneity(const ModelPtr &model);
```

Comment:
Brief should start with a capital letter, like any sentence.  There should also be a full explanation given in the lines below.